src/app/Archives/AllArchivedRecordingsTable.tsx

```diff
@@ -240,25 +240,34 @@ export const AllArchivedRecordingsTable: React.FunctionComponent<AllArchivedReco
     );
   };
 
-  const handleDeleteButton = () => {
+  const handleDeleteButton = React.useCallback(() => {
     if (context.settings.deletionDialogsEnabledFor(DeleteWarningType.DeleteArchivedRecordings)) {
       setWarningModalOpen(true);
     }
     else {
       handleDeleteRecordings();
     }
-  }
+  }, [handleDeleteRecordings])
```

Comment:
`context`, `context.settings`, and `setWarningModalOpen` should all also be dependencies here.